systems/trajectory_optimization/dircon_opt_constraints.h

```diff
@@ -69,15 +70,17 @@ class DirconDynamicConstraint : public DirconAbstractConstraint<T> {
   DirconDynamicConstraint(const drake::multibody::MultibodyPlant<T>& plant,
                           DirconKinematicDataSet<T>& constraints,
                           int num_positions, int num_velocities, int num_inputs,
-                          int num_kinematic_constraints);
+                          int num_kinematic_constraints_wo_skipping,
+                          int num_quat_slack);
```

Comment:
`num_quat_slack` is the dimension of the slack variable for quaternion. It's always 1 if the MBP is quaternion floating based. Otherwise, it's 0. I'll add documentation about this.